flow/project.py

```diff
@@ -3613,7 +3616,7 @@ def submit(
                 "The 'names' argument must be a sequence of strings, however "
                 f"a single string was provided: {names}."
             )
-        if walltime is not None:
+        if walltime is not None and not isinstance(walltime, datetime.timedelta):
```

Comment:
Since we're considering handling the walltime requested via directives, I propose deprecating the walltime argument from the public api.

Reply:
Yes, that is probably the right idea.